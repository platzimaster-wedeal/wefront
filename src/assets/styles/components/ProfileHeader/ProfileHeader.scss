@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../Breakpoints.scss';

.profile-header {
  position: relative;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: auto;

  &__description {
    margin-top: $space-size-extra-small - 6;
    @include font-size($font-size-small);
    color: $color-black-90;

    @include media(tablet) {
      @include font-size($font-size-normal);
    }
  }

  &__profession {
     @include font-size($font-size-normal);
    color: $color-black-60;

    @include media(tablet) {
      @include font-size($font-size-medium);
    }
  }

  &__actions {
    display: flex;
    flex-flow: row nowrap;
    width: 100%;
    justify-content: center;
    align-items: center;
    
    & button {
      box-sizing: border-box;
      margin: $space-size-small;
    }
  }
  
}