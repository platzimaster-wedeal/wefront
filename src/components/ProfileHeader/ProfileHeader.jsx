import React from "react";

// Styles
import "../../assets/styles/components/ProfileHeader/ProfileHeader.scss";

// Components
import ProfileHeaderPhoto from "../ProfileHeaderPhoto/ProfileHeaderPhoto";
import ProfileHeaderAction from "../ProfileHeaderActions/ProfileHeaderActions";

const ProfileHeader = ({ description, profession }) => {
 return (
  <section className="profile-header">
   <ProfileHeaderPhoto />
   <small className="profile-header__description">
    {description && description}
   </small>
   <strong className="profile-header__profession">
    {profession && profession}
   </strong>

   <div className="profile-header__actions">
    <ProfileHeaderAction isVisitor={false} />
   </div>
  </section>
 );
};

export default ProfileHeader;
