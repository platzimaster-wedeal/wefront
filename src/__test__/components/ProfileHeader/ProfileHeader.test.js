import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import ProfileHeader from "../../../components/ProfileHeader/ProfileHeader";

describe("Test for the component ProfileHeader :D", () => {
 const ProfileHeaderMounted = mount(<ProfileHeader />);
 const ProfileHeaderCreated = create(<ProfileHeader />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(ProfileHeaderCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(ProfileHeaderMounted.length).toEqual(1);
  });
  test("Should the description of the User must be a string", () => {
   expect(
    typeof ProfileHeaderMounted.find(".profile-header__description").text()
   ).toEqual("string");
  });
  test("Should the profession of the User must be a string", () => {
   expect(
    typeof ProfileHeaderMounted.find(".profile-header__profession").text()
   ).toEqual("string");
  });
 });
});
