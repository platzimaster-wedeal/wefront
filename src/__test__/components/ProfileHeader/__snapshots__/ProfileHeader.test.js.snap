// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component ProfileHeader :D Visual tests for the component should the comopnent have the same tree 1`] = `
<section
  className="profile-header"
>
  <div
    className="profile-header-photo"
  >
    <figure
      className="profile-header-photo__figure"
    >
      <img
        alt="Tupac Shakur Walks"
        aria-label="Tupac Shakur Walks"
        src={Object {}}
      />
      <button
        className="button-actionSA"
        style={
          Object {
            "fontSize": "70px",
            "height": "30px",
            "width": "30px",
          }
        }
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3.2"
          />
          <path
            d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
          />
        </svg>
      </button>
    </figure>
    <h2
      className="profile-header-photo__name"
    >
      Tupac Shakur Walks
    </h2>
  </div>
  <small
    className="profile-header__description"
  />
  <strong
    className="profile-header__profession"
  />
  <div
    className="profile-header__actions"
  >
    <div
      className="profile-header__actions"
    >
      <button
        className="button button__primary"
      >
         
        Edit
         
      </button>
       
    </div>
  </div>
</section>
`;
